### DISCUSSION TITLE
Enchancement Idea for Documentation: Enable Tags in documentation which would be searchable on platform.uno to improve searchresults

### DISCUSSION THREAD

**Discussion by DevTKSS** [unoplatform/uno#18809] [Ideas]
Hey there!
I am quite new to all of the uno stuff but eager to learn it. So, since I am researching many different topics in the documentation on plattform.uno, I find it quite hard to find what I am searching for in the particular moment. E.g I do not know the right keyword or the even if I know them, the articles related to them are sometimes (quite often to be honest) kind of everywhere in this big (thats positiv meant!) documentation.

For Example:
I was reserching some time to navigation. So I did find the Dialog things and a how to for the flyout and modal one. But what I actually did not find were just sample apps to all of the different possibilitys to better understand this.

I did not understand how it could be, that there seems to be no sample on this except from this very articles in documentation with the short snippets.

Until Today, where I did research for another topic, where was a sample app in this article linked (Web Api consume). Since you might know, a API-Key which is needed for this should be stored secretly and not clear typed in the class, so I thought about the Configuration Extension of Uno, but beside of the UnoFeature to add and the thing in app OnLaunched which is described, My Editor is not finding the "CreateBuilder" which is needed for this. So I was searching now for some sample which should show me what is missing in my uno app and scrolled in the uno.samples repo.
Hell seems like I had tomatos on my eyes, because there are so many samples which I did miss.

Okay, so what if other bloody newbies are like me and not finding them because, maybe not familiar with github or did not click through every single sample app which is listed in the section in documentation. Name me stupid not to do this, but when I am searching for e.g. navigation and some description I roughly readed is not naming this and instead something related which word I maybe not put into relation to this, I will oversee this!

So from this Thought my improvement Idea for you:
Would there be a possibility, to add keywords in form of maybe tags into the documentation?
So even if the writer would just want to give a short description of something, or even the reader is just roughly reading like me sometimes, this tags could be listed in side panel below the Edit this page or below the headlines also!
Then maybe via the search bar on top the search engine on plattform.uno could maybe also lookup this tags for guiding the searching person to this article, even if it is not naming this directly. So lets say a sample description telling it is about mvux and pagination and many other things. A reader like me would not know that pagination might be a part of navigation, but the search engine would list this refering to this tag for me and the chance to find this and so I would more likly look it up, that would be much higher!

Hope I did explain it clear enough. On Questions to this please ask!

**Comment by DevTKSS** [comment]
@Jen-Uno as you are checking out docs improvements, what about this idea? Maybe could work with yaml markdig extensionn...

**Reply by Jen-Uno** [reply]
@DevTKSS Hi! I am! I have asked @booskander to review the PRs you submitted. I will get to this one soon :)

**Reply by DevTKSS** [reply]
@Jen-Uno @booskander I am currently reading through the docfx documentation for my own docs and got curious, as I readed about [xrefmap](https://dotnet.github.io/docfx/docs/links-and-cross-references.html#cross-reference-using-xrefmap) and remembered that your ci and the import_external_docs.ps1 scripts are... doing exactly that just with additional effort?
I dont expect you to not know about and this process we are using currently is well thought, so maybe someone can tell why is this that way?
I mean, what I would understand this xrefmap to maybe enable is, that all repos e.g. Uno has, could have their own docfx file, would enable everyone, like also contributors like me, to "Just" run the docs deployment using docfx locally without need the whole Uno.UI Repo, no Headache with the UWPGenerator (didnt Uno moved on from UWP? Why do we use it still?) that is eitherway failing *maybe because of the docfx.console is depreciated and not longer available?* and if Uno then deploys to publish the docs from their side, they would only have to source the xrefmap files from the repos to include 🚀

To me this does sound like very fewer work and better maintainabillity of each of the docs! Would appreciate to hear your opinion!

**Comment by DevTKSS** [comment]
@Jen-Uno  not sure but I thing I've seen in a PR in extensions repo the team using tags in docs yaml headers already 🤔
Do you maybe have a good news about this to now be possible? If yes, it would be cool to have this introduced in the Contributing docs too 😍
